lms/djangoapps/courseware/model_data.py

```diff
@@ -275,28 +275,26 @@ def find_or_create(self, key):
             return field_object
 
         if key.scope == Scope.user_state:
-            field_object, __ = StudentModule.objects.get_or_create(
+            field_object = StudentModule(
                 course_id=self.course_id,
                 student_id=key.user_id,
                 module_state_key=key.block_scope_id,
-                defaults={
-                    'state': json.dumps({}),
-                    'module_type': key.block_scope_id.block_type,
-                },
+                state=json.dumps({}),
+                module_type=key.block_scope_id.block_type,
             )
```

Comment:
I think you just have to manually do the get, catch the exception, and return a newly constructed `StudentModule` from that?